app/src/main/java/ar/com/wolox/android/cookbook/room/RoomRecipePresenter.kt

```diff
@@ -0,0 +1,85 @@
+package ar.com.wolox.android.cookbook.room
+
+import android.app.Application
+import android.os.Handler
+import ar.com.wolox.android.cookbook.room.database.RoomDataEntity
+import ar.com.wolox.android.cookbook.room.database.RoomDatabaseManager
+import ar.com.wolox.wolmo.core.presenter.BasePresenter
+import javax.inject.Inject
+
+class RoomRecipePresenter @Inject constructor(
+    val application: Application
+) : BasePresenter<RoomRecipeView>() {
+
+    private lateinit var db: RoomDatabaseManager
+    private var userName: String? = null
+    private val handler = Handler()
+
+    override fun onViewAttached() {
+        super.onViewAttached()
+        db = RoomDatabaseManager.invoke(application.baseContext)
+    }
+
+    fun onSessionButtonClicked(user: String) {
+        userName?.let {
+            userName = null
+            view.logout()
+        } ?: run {
+            if (user.isNotEmpty()) {
+                userName = user
+                Thread(Runnable {
```

Comment:
Using threads isn't the most elegant solution to this, it could generate some troubles. I think there should be a way to handle it with callbacks or you can do it with RxJava. Even though which I recommend is to use Coroutines which are also using Flor and Doni, so I also recommend to talk about this with them.